### PR TITLE
Update Git instructions & add ui testing documentation

### PR DESCRIPTION
**Pull request by marc1706**
Some of the git documentation was a bit outdated and documentation on UI tests was missing completely. Also fixed a few build issues.

### CODE REVIEW

File: development/development/git.rst [hunk 1]
```diff
@@ -357,7 +380,7 @@ Merging to 3.0.x
 ^^^^^^^^^^^^^^^^
 .. warning::
 
-    **ALL** merges to 3.0.x **must** also be merged to 3.1.x and master!
+    **ALL** merges to 3.0.x **must** also be merged to 3.1.x, 3.2.x, and master!
```
Comment (VSEphpbb) [inline development/development/git.rst]: Do we even need the Merging to 3.0.x anymore, really?

Comment (marc1706) [reply development/development/git.rst]: Could be removed, left it in there for now.

**Comment by VSEphpbb** [comment]
I say dump it. 3.0 is so 5 years ago :P

File: development/development/git.rst (hunk 1 above)
Comment (VSEphpbb) [reply development/development/git.rst]: I say dump it. 3.0 is so 5 years ago :P

Comment (marc1706) [reply development/development/git.rst]: Ok 🦄

**Comment by marc1706** [comment]
Removed the 3.0 merge instructions.

**Comment by VSEphpbb** [comment]
@marc1706 a year later...what's the status on this PR?

**Comment by marc1706** [comment]
I'll try to revisit this the next few days to finish this finally.

**Comment by marc1706** [comment]
@VSEphpbb updated

**Comment by VSEphpbb** [comment]
Is this being merged to the right branch?

File: development/development/coding_guidelines.rst [hunk 2]
```diff
@@ -5,8 +5,8 @@ phpBB's coding guidelines are maintained in its
 `source code repository <https://github.com/phpbb/phpbb>`_. You can find the
 latest versions on area51:
 
-* Rules for `Olympus (3.0.x) code <http://area51.phpbb.com/docs/30x/coding-guidelines.html>`_
-* Rules for `Ascraeus (3.1.x) code <http://area51.phpbb.com/docs/31x/coding-guidelines.html>`_
+* Rules for `Olympus (3.0.x) code <https://area51.phpbb.com/docs/30x/coding-guidelines.html>`_
+* Rules for `Ascraeus (3.1.x) code <https://area51.phpbb.com/docs/31x/coding-guidelines.html>`_
```
Comment (Crizz0) [inline development/development/coding_guidelines.rst]: Maybe the right time to remove 3.0 and 3.1 and add 3.2 and 3.3. instead?

Comment (marc1706) [reply development/development/coding_guidelines.rst]: Well, this PR is still agains 3.1.x so maybe as another task after merging this

**Comment by marc1706** [comment]
@VSEphpbb we'll have to do a few more updates for 3.3.x and 3.2.x respectively

**Pull request merged**